src/test/java/uk/gov/dwp/dataworks/DataKeyServiceApplicationTests.java

```diff
@@ -1,15 +1,126 @@
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.dwp.dataworks.dto.DecryptDataKeyResponse;
+import uk.gov.dwp.dataworks.dto.GenerateDataKeyResponse;
+import uk.gov.dwp.dataworks.errors.CurrentKeyIdFailure;
+import uk.gov.dwp.dataworks.errors.DataKeyDecryptionFailure;
+import uk.gov.dwp.dataworks.errors.DataKeyGenerationFailure;
+import uk.gov.dwp.dataworks.errors.GarbledDataKeyError;
+import uk.gov.dwp.dataworks.provider.CurrentKeyIdProvider;
+import uk.gov.dwp.dataworks.provider.DataKeyDecryptionProvider;
+import uk.gov.dwp.dataworks.provider.DataKeyGeneratorProvider;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @RunWith(SpringRunner.class)
-@SpringBootTest
+@SpringBootTest()
+@ActiveProfiles("Test")
+@AutoConfigureMockMvc
 public class DataKeyServiceApplicationTests {
 
+    @Autowired
+    private CurrentKeyIdProvider currentKeyIdProvider;
+
+    @Autowired
+    private DataKeyGeneratorProvider dataKeyGeneratorProvider;
+
+    @Autowired
+    private DataKeyDecryptionProvider dataKeyDecryptionProvider;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Before
+    public void setup() {
+        Mockito.reset(currentKeyIdProvider, dataKeyGeneratorProvider, dataKeyDecryptionProvider);
+    }
+
     @Test
     public void contextLoads() {
     }
+
+    @Test
+    public void pingWorks() throws Exception {
+        mockMvc.perform(get("/ping"))
+                .andExpect(status().isOk())
+                .andExpect(content().string(""));
+    }
+
+    @Test
+    public void generateDataKeyRequest() throws Exception {
+        given(currentKeyIdProvider.getKeyId()).willReturn("myKeyId");
+
+        GenerateDataKeyResponse response = new GenerateDataKeyResponse("encryptionId", "plainKey", "cipherKey");
+
+        given(dataKeyGeneratorProvider.generateDataKey("myKeyId"))
+                .willReturn(response);
+
+        mockMvc.perform(get("/datakey"))
+                .andExpect(status().isCreated())
+                .andExpect(content().json(new ObjectMapper().writeValueAsString(response)));
+
+        // Must also allow a trailing slash
+        mockMvc.perform(get("/datakey/"))
```

Comment:
Would generally try to keep one test to one functional check - i.e. have a different test for the second call.